src/main/java/com/revature/assignforce/beans/Settings.java

```diff
@@ -1,175 +1,212 @@
-		this.alias = alias;
-	}
-
-	public Integer getTrainersPerPage() {
-		return trainersPerPage;
-	}
-
-	public void setTrainersPerPage(Integer trainersPerPage) {
-		this.trainersPerPage = trainersPerPage;
-	}
-
-	public Integer getReportGrads() {
-		return reportGrads;
-	}
-
-	public void setReportGrads(Integer reportGrads) {
-		this.reportGrads = reportGrads;
-	}
-
-	public Integer getBatchLength() {
-		return batchLength;
-	}
-
-	public void setBatchLength(Integer batchLength) {
-		this.batchLength = batchLength;
-	}
-
-	public Integer getReportIncomingGrads() {
-		return reportIncomingGrads;
-	}
-
-	public void setReportIncomingGrads(Integer reportIncomingGrads) {
-		this.reportIncomingGrads = reportIncomingGrads;
-	}
-
-	public Integer getMinBatchSize() {
-		return minBatchSize;
-	}
-
-	public void setMinBatchSize(Integer minBatchSize) {
-		this.minBatchSize = minBatchSize;
-	}
-
-	public Integer getMaxBatchSize() {
-		return maxBatchSize;
-	}
-
-	public void setMaxBatchSize(Integer maxBatchSize) {
-		this.maxBatchSize = maxBatchSize;
-	}
-
-	public Integer getTrainerBreakDays() {
-		return trainerBreakDays;
-	}
-
-	public void setTrainerBreakDays(Integer trainerBreakDays) {
-		this.trainerBreakDays = trainerBreakDays;
-	}
-
-	public Integer getDefaultLocation() {
-		return defaultLocation;
-	}
-
-	public void setDefaultLocation(Integer defaultLocation) {
-		this.defaultLocation = defaultLocation;
-	}
-
-	public Integer getDefaultBuilding() {
-		return defaultBuilding;
-	}
-
-	public void setDefaultBuilding(Integer defaultBuilding) {
-		this.defaultBuilding = defaultBuilding;
-	}
-
-	public String getDefaultNamePattern() {
-		return defaultNamePattern;
-	}
-
-	public void setDefaultNamePattern(String defaultNamePattern) {
-		this.defaultNamePattern = defaultNamePattern;
-	}
-
-}
+package com.revature.assignforce.beans;
+
+import java.io.Serializable;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+
+import com.revature.assignforce.validators.IsValidInterval;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import javax.validation.constraints.Min;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * SVP settings for the angular application. As of 8/4/2018 the angular form 
+ * cannot be submitted for settings unless all integer fields are filled.
+ * defaultNamePattern can be submitted as an empty string but not null.
+ * https://github.com/revaturelabs/assignforce-client/wiki/Settings
+ * https://github.com/revaturelabs/assignforce-client/wiki/RESTful-API#settings-service
+ */
+@Component
+@Entity
+@Table(name = "Settings")
+@IsValidInterval(startInterval="minBatchSize", endInterval="maxBatchSize", 
+                 inclusive=true, ifNull=false)

```

Comment:
Batch size isn't an interval, so we need to modify the naming of this validation method (Which should just be "Interval" regardless), or use some other method to validate this.

Reply:
Changed name of IsValidInterval to IntervalHolder.